File: single-language-analysis/dataset-files/python/test_smell_024.py
import pytest

class TestArrayInversion:
    @pytest.mark.skip(reason="coz it's not quick todo: to review")
    def test_smell_024(self):
        file = "resources/IntegerArray.txt"
        try:
            list_ = MyArrayUtil.list_integer_from_file(file)

            A = MyArrayUtil.list_integers_to_integer_array(list_)

            D_total = ArrayInversion.sort_and_count(A, len(A), 0, len(A) - 1)

            D = D_total.get_left()

            MyArrayUtil.check_array_integer_sorted(D)

            expected_inversions = ArrayInversion.big_integer_brute_force_count(A)
            print("expect " + str(expected_inversions) + " inversions")
            assert expected_inversions == int(D_total.get_right())

        except FileNotFoundError as e:
            print(e)
            assert False
        except IOError as e:
            print(e)
            assert False

File: single-language-analysis/dataset-files/java/TestSmell024.java
// Original URL: https://github.com/henrytxz/Algo/blob/f0b186aa128ebd01835013f432b793593c4a0e6d/slowTests/not_quick/ArrayInversionTest.java#L82-L108

public class TestSmell024 {
    @Ignore //coz it's not quick todo: to review
    @Test
    public void testBigIntegerSortAndCount() throws Exception {
        File file = new File("resources/IntegerArray.txt");
        try {
            List<Integer> list = MyArrayUtil.ListIntegerFromFile(file);

            Integer[] A = MyArrayUtil.ListIntegersToIntegerArray(list);

            Pair<Integer[],Integer> D_total = ArrayInversion.SortAndCount(A, A.length, 0, A.length-1);

            Integer[] D = D_total.getLeft();

            MyArrayUtil.checkArrayIntegerSorted(D);

            BigInteger expectedInversions = ArrayInversion.BigInteger_bruteForceCount(A);
            System.out.println("expect "+expectedInversions+" inversions");
//            Assert.assertEquals(expectedInversions, (int) D_total.getRight());

        } catch (FileNotFoundException e) {
            e.printStackTrace();  //To change body of catch statement use File | Settings | File Templates.
            Assert.fail();
        } catch (IOException e) {
            e.printStackTrace();
            Assert.fail();
        }
    }
}
